File: NUTENC_CS/frmAlunoLINQToSQL.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.SqlClient;
using System.Windows.Forms;

namespace NUTENC_CS
{
    public partial class frmAlunoLINQToSQL : Form
    {
        public frmAlunoLINQToSQL()
        {
            InitializeComponent();
        }

        private void btnIncluir_Click(object sender, EventArgs e)
        {
            BancoDeDadosDataContext banco = new BancoDeDadosDataContext();

            Alunos aluno = new Alunos();

            aluno.IdAluno = System.Convert.ToInt32(txtIdAluno.Text);
            aluno.Nome = txtNome.Text;
            aluno.Endereco = txtEndereco.Text;
            aluno.Cidade = txtCidade.Text;
            aluno.Telefone = txtTelefone.Text;
            aluno.Email = txtEmail.Text;

            banco.Alunos.InsertOnSubmit(aluno);

            banco.SubmitChanges();

            banco.Connection.Close();


            txtIdAluno.Text = "";
            txtNome.Text = "";
            txtEndereco.Text = "";
            txtCidade.Text = "";
            txtTelefone.Text = "";
            txtEmail.Text = "";

            MessageBox.Show("Aluno inserido com sucesso!", "Inserir aluno");
        }

        private void btnAtualizar_Click(object sender, EventArgs e)
        {
            {
                {
                    BancoDeDadosDataContext banco = new BancoDeDadosDataContext();

                    var aluno = banco.Alunos
                                .Where(a => a.IdAluno.ToString() == txtIdAluno.Text)
                                .SingleOrDefault();


                    aluno.IdAluno = Convert.ToInt32(txtIdAluno.Text);
                    aluno.Nome = txtNome.Text;
                    aluno.Endereco = txtEndereco.Text;
                    aluno.Cidade = txtCidade.Text;
                    aluno.Telefone = txtTelefone.Text;
                    aluno.Email = txtEmail.Text;

                    banco.SubmitChanges();

                    banco.Connection.Close();
                }


                txtIdAluno.Text = "";
                txtNome.Text = "";
                txtEndereco.Text = "";
                txtCidade.Text = "";
                txtTelefone.Text = "";
                txtEmail.Text = "";

                MessageBox.Show("Aluno atualizado com sucesso!", "Atualizar aluno");
            }

        }

        private void btnExcluir_Click(object sender, EventArgs e)
        {
            {
                BancoDeDadosDataContext banco = new BancoDeDadosDataContext();

                var aluno = banco.Alunos
                                .Where(a => a.IdAluno.ToString() == txtIdAluno.Text)
                                .SingleOrDefault();

                banco.Alunos.DeleteOnSubmit(aluno);

                banco.SubmitChanges();

                banco.Connection.Close();

                txtIdAluno.Text = "";
                txtNome.Text = "";
                txtEndereco.Text = "";
                txtCidade.Text = "";
                txtTelefone.Text = "";
                txtEmail.Text = "";

                MessageBox.Show("Aluno excluído com sucesso!", "Excluir aluno");
            }


        }

        private void btnPesquisar_Click(object sender, EventArgs e)
        {
            {
                BancoDeDadosDataContext banco = new BancoDeDadosDataContext();

                var alunos = banco.Alunos
                             .Where(a => a.Nome.Contains(txtNome.Text))
                             .ToList();

                dgvListaDeAlunos.DataSource = alunos;
                dgvListaDeAlunos.Refresh();

                banco.Connection.Close();

            }

        }

        private void dgvListaDeAlunos_DoubleClick(object sender, EventArgs e)
        {
            this.txtIdAluno.Text = dgvListaDeAlunos.CurrentRow.Cells["colIdAluno"].Value.ToString();
            this.txtNome.Text = dgvListaDeAlunos.CurrentRow.Cells["colNome"].Value.ToString();
            this.txtEndereco.Text = dgvListaDeAlunos.CurrentRow.Cells["colEndereco"].Value.ToString();
            this.txtCidade.Text = dgvListaDeAlunos.CurrentRow.Cells["colCidade"].Value.ToString();
            this.txtTelefone.Text = dgvListaDeAlunos.CurrentRow.Cells["colTelefone"].Value.ToString();
            this.txtEmail.Text = dgvListaDeAlunos.CurrentRow.Cells["colEmail"].Value.ToString();
        }

        private void btnLimpar_Click(object sender, EventArgs e)
        {
            txtIdAluno.Text = "";
            txtNome.Text = "";
            txtEndereco.Text = "";
            txtCidade.Text = "";
            txtTelefone.Text = "";
            txtEmail.Text = "";
        }
    }
}

File: NUTEC/frmAlunoLINQToSQL.vb
Public Class frmAlunoLINQToSQL
    Private Sub btnIncluir_Click(sender As Object, e As EventArgs) Handles btnIncluir.Click

        Dim banco As New BancoDeDadosDataContext

        Dim aluno As New Alunos

        aluno.IdAluno = txtIdAluno.Text
        aluno.Nome = txtNome.Text
        aluno.Endereco = txtEndereco.Text
        aluno.Cidade = txtCidade.Text
        aluno.Telefone = txtTelefone.Text
        aluno.Email = txtEmail.Text

        banco.Alunos.InsertOnSubmit(aluno)

        banco.SubmitChanges()

        banco.Connection.Close()


        txtIdAluno.Text = ""
        txtNome.Text = ""
        txtEndereco.Text = ""
        txtCidade.Text = ""
        txtTelefone.Text = ""
        txtEmail.Text = ""

        MessageBox.Show("Aluno inserido com sucesso!", "Inserir aluno")

    End Sub

    Private Sub btnAtualizar_Click(sender As Object, e As EventArgs) Handles btnAtualizar.Click
        Dim banco As New BancoDeDadosDataContext

        Dim aluno = (From a In banco.Alunos Where a.IdAluno = txtIdAluno.Text Select a).SingleOrDefault

        aluno.IdAluno = txtIdAluno.Text
        aluno.Nome = txtNome.Text
        aluno.Endereco = txtEndereco.Text
        aluno.Cidade = txtCidade.Text
        aluno.Telefone = txtTelefone.Text
        aluno.Email = txtEmail.Text

        banco.SubmitChanges()

        banco.Connection.Close()

        txtIdAluno.Text = ""
        txtNome.Text = ""
        txtEndereco.Text = ""
        txtCidade.Text = ""
        txtTelefone.Text = ""
        txtEmail.Text = ""

        MessageBox.Show("Aluno atualizado com sucesso!", "Atualizar aluno")

    End Sub

    Private Sub btnExcluir_Click(sender As Object, e As EventArgs) Handles btnExcluir.Click
        Dim banco As New BancoDeDadosDataContext

        Dim aluno = (From a In banco.Alunos Where a.IdAluno = txtIdAluno.Text Select a).SingleOrDefault

        banco.Alunos.DeleteOnSubmit(aluno)

        banco.SubmitChanges()

        banco.Connection.Close()

        txtIdAluno.Text = ""
        txtNome.Text = ""
        txtEndereco.Text = ""
        txtCidade.Text = ""
        txtTelefone.Text = ""
        txtEmail.Text = ""

        MessageBox.Show("Aluno excluído com sucesso!", "Excluir aluno")

    End Sub

    Private Sub btnPesquisar_Click(sender As Object, e As EventArgs) Handles btnPesquisar.Click



        Dim banco As New BancoDeDadosDataContext

        Dim alunos = (From a In banco.Alunos Where a.Nome.Contains(txtNome.Text) Select a).ToList


        dgvListaDeAlunos.DataSource = alunos
        dgvListaDeAlunos.Refresh()

        banco.Connection.Close()

    End Sub

    Private Sub dgvListaDeAlunos_DoubleClick(sender As Object, e As EventArgs) Handles dgvListaDeAlunos.DoubleClick
        Me.txtIdAluno.Text = dgvListaDeAlunos.CurrentRow.Cells("colIdAluno").Value
        Me.txtNome.Text = dgvListaDeAlunos.CurrentRow.Cells("colNome").Value
        Me.txtEndereco.Text = dgvListaDeAlunos.CurrentRow.Cells("colEndereco").Value
        Me.txtCidade.Text = dgvListaDeAlunos.CurrentRow.Cells("colCidade").Value
        Me.txtTelefone.Text = dgvListaDeAlunos.CurrentRow.Cells("colTelefone").Value
        Me.txtEmail.Text = dgvListaDeAlunos.CurrentRow.Cells("colEmail").Value
    End Sub

    Private Sub btnLimpar_Click(sender As Object, e As EventArgs) Handles btnLimpar.Click
        txtIdAluno.Text = ""
        txtNome.Text = ""
        txtEndereco.Text = ""
        txtCidade.Text = ""
        txtTelefone.Text = ""
        txtEmail.Text = ""
    End Sub
End Class
